app/lib/identity_tijuana/member_sync.rb

```diff
@@ -109,15 +109,20 @@ def self.get_id_change_date(member, field_info_type, default_change_date = nil)
       end
       # Iterate through the relevant audit log list, moving backwards in time
       # until we find a change to the field or fields in question.
-      member.send(audit_list_name).where(auditable_type: auditable_type).reverse_each do |audit|
+      # Note that we use `reorder` to remove all prior ordering clauses,
+      # as Audited injectes `order by version asc` when order clause is used.
+      member.send(audit_list_name)
+            .where(auditable_type: auditable_type)
+            .reorder(created_at: :desc)
+            .each do |audit|
         # Check ancillary matching field, where required.
         if !ancillary_match_value.nil?
           case auditable_type
           when 'PhoneNumber'
             phone_number_type =
               audit.audited_changes['phone_type'] ||
               PhoneNumber.find_by(id: audit.auditable_id).try(:phone_type)
-            next unless phone_number_type == ancillary_match_value
+            next unless phone_number_type.to_sym == ancillary_match_value
```

Comment:
Is `ancillary_match_value` always a symbol when not nil? From the places its set above, it looks like it could also sometimes be a string?

Reply:
NB I don't know if Ruby returns true or false for `"foo".to_sym == "foo"`, but if false maybe this line isn't needed at all?

Reply:
> Is ancillary_match_value always a symbol when not nil?

No, it is only a symbol when `auditable_type` is `PhoneNumber` -> `ancillary_match_value = field_info_type` with call eg. `get_id_change_date(member, :mobile, id_mobile&.updated_at || member&.updated_at)`.
Otherwise, it's integer eg. `ancillary_match_value = Settings.tijuana.email_subscription_id`

> NB I don't know if Ruby returns true or false for "foo".to_sym == "foo", but if false maybe this line isn't needed at all?

`"foo".to_sym == "foo"` is `false`, but with `PhoneNumber` case we want to test `"foo".to_sym == :foo`, as in
`"mobile".to_sym == :mobile`.

Reply:
Again, I'm not sure if `1 == "1"` will eval to true or false in ruby, but I hope the latter. In which case the tests may well be behaving differently to runtime when integrated with Id.

Reply:
ahh, I totally missed that! 
well, yes `1 == "1"` is `false`, we should really cast these `audit_subscription_id` and `ancillary_match_value` to int when the `auditable_type` is `MemberSubscription`